### PR TITLE
Improve testing in pyscalcli, remove non-triggered code

### PR DESCRIPTION
**Pull request by berland**

### CODE REVIEW

File: pyscal/pyscalcli.py [hunk 1]
```diff
@@ -205,21 +204,7 @@ def pyscal_main(
...
-        int_param_go is not None
-        and isinstance(int_param_go, list)
-        and len(int_param_go) > 1
-    ):
-        warnings.warn(
-            "SATNUM specific interpolation parameters are deprecated in "
-            "the pyscal command line client. "
-            "Use interp_relperm from subscript or the API directly",
-            FutureWarning,
-        )
```
Comment (berland) [inline pyscal/pyscalcli.py:222]: This is impossible to reach, as `argparse` bails out before this.

File: pyscal/pyscalcli.py [hunk 2]
```diff
@@ -248,13 +233,6 @@ def pyscal_main(
             scalinput_df, h=delta_s
         )  # can be both water-oil, water-oil-gas, or gas-water
 
-    if (
-        int_param_wo is not None or int_param_go is not None
-    ) and "CASE" not in scalinput_df:
-        raise ValueError(
-            "Interpolation parameter provided but no CASE column in input data"
-        )
-
```
Comment (berland) [inline pyscal/pyscalcli.py:257]: This error scenario is detected already on `load_relperm_df()`

File: tests/test_pyscalcli.py [hunk 3]
```diff
@@ -17,9 +17,6 @@ def test_installed():
     """Test that the command line client is installed in PATH and
     starts up nicely"""
     assert subprocess.check_output(["pyscal", "-h"])
-    assert subprocess.check_output(["pyscal", "--help"])
-
-    assert subprocess.check_output(["pyscal", "--version"])
```
Comment (berland) [inline tests/test_pyscalcli.py:22]: These lines do not give any testing value, they only take a lot of time to execute.

File: pyscal/pyscalcli.py (hunk 1 above)
Comment (lars-petter-hauge) [reply pyscal/pyscalcli.py:222]: Noting the slight inconsistency that the description states this function is available mainly to be used _without_ argparse.. Not sure if this should be removed if it is intended to be used from other places..

Comment (berland) [reply pyscal/pyscalcli.py:222]: Thank you for reading the comments in the code. You are completely right.

File: pyscal/pyscalcli.py (hunk 2 above)
Comment (berland) [reply pyscal/pyscalcli.py:257]: Reinstated as it turned out this scenario was indeed possible to reach.

File: pyscal/pyscalcli.py (hunk 1 above)
Comment (berland) [reply pyscal/pyscalcli.py:223]: You are completely right, this check should be there, and tested.

File: pyscal/pyscalcli.py (hunk 2 above)
Comment (berland) [reply pyscal/pyscalcli.py:258]: Reinstated after understanding fully what this check did. Test added.

**Review by lars-petter-hauge — APPROVED** [review]

**Pull request closed without merging**